tests/test_api/test_benefits.py

```diff
@@ -0,0 +1,108 @@
+        "strategy": "elevate_comb_correct",
+        "projection": "all_projections",
+        "future_year": "two thousand eighty",
+        "current_situation": {
+            "projection": database.site.attrs.benefits.current_projection,
+            "year": database.site.attrs.benefits.current_year,
+        },
+        "baseline_strategy": database.site.attrs.benefits.baseline_strategy,
+        "discount_rate": 0.07,
+    }
+
+    # When user presses add benefit and chooses the values
+    with pytest.raises(ValueError):
+        # Assert error if the event is not a correct value
+        benefit = api_benefits.create_benefit(benefit_dict, database)
+    # correct value
+    benefit_dict["future_year"] = 2080
+    benefit = api_benefits.create_benefit(benefit_dict, database)
+
+    with pytest.raises(ValueError):
+        # Assert error if name already exists
+        api_benefits.save_benefit(benefit, database)
+
+    # Change name to something new
+    benefit_dict["name"] = "benefit_raise_properties_2080"
+
+    # When user presses "Check scenarios" the object will be created in memory and a dataframe will be returned
+    benefit = api_benefits.create_benefit(benefit_dict, database)
+    df = api_benefits.check_benefit_scenarios(benefit, database)
+    if sum(df["scenario created"] == "No") > 0:
+        api_benefits.create_benefit_scenarios(benefit, database)
+
+    df = api_benefits.check_benefit_scenarios(benefit, database)
+    if sum(df["scenario created"] == "No") == 0:
+        api_benefits.save_benefit(benefit, database)
+
+    with pytest.raises(RuntimeError):
+        # Assert error if not yet run
+        api_benefits.run_benefit("benefit_raise_properties_2080", database)
+
+    # Simulate FIAT run by creating the results folder for each run and adding a csv with a total EAD
+    EADs = pd.Series(
+        {
+            "current_no_measures": 10000000,
+            "current_with_strategy": 3000000,
+            "future_no_measures": 16000000,
+            "future_with_strategy": 5000000,
+        }
+    )
+
+    results_path = test_database_path.joinpath("charleston", "output", "results")
+    for index, scenario in benefit.scenarios.iterrows():
+        res_scn_path = results_path.joinpath(scenario["scenario created"]).joinpath(
+            "fiat_model"
+        )
+        res_scn_path.mkdir(parents=True)
+
+        df = pd.DataFrame({"EAD": EADs[index]}, index=[0])
+        df.to_csv(res_scn_path.joinpath("metrics.csv"), index=False)
+
+    api_benefits.run_benefit("benefit_raise_properties_2080", database)
+
+    # If the user edits the benefit to add costs
+    benefit_dict_new = benefit_dict.copy()
+    benefit_dict_new["implementation_cost"] = 30000000
+    benefit_dict_new["annual_maint_cost"] = 0
+    benefit = api_benefits.create_benefit(benefit_dict_new, database)
+    api_benefits.edit_benefit(benefit, database)
+    api_benefits.run_benefit("benefit_raise_properties_2080", database)
```

Comment:
any assertion to check if it evaluated the costs now? I.e. cost-benefits lower than before

Reply:
added some tests